parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -471,6 +471,10 @@ pub mod pallet {
 				&update.sync_aggregate,
 				update.signature_slot,
 			)?;
+			ensure!(
+				update.finalized_header.slot > Self::latest_finalized_header().beacon_slot,
+				Error::<T>::DuplicateFinalizedHeaderUpdate
+			);
 			let finalized_block_root = Self::verify_finalized_header(
 				&update.attested_header,
 				&update.finalized_header,
```

Comment:
In commit https://github.com/Snowfork/snowbridge/pull/835/commits/faa4526f342ee190f885f945f80d4aa2d1f6357a
just move this check out from `verify_finalized_header`, because it does not apply to `process_sync_committee_period_update`

Reply:
Good catch Ron 🙌 

I see that in the consensus spec for [validate_light_client_update](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#validate_light_client_update), they don't have such a check either.

Instead in [apply_light_client_update](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#apply_light_client_update), the finalized header is only applied if it is newer than the latest finalized header in storage. So we should probably follow the same approach if we decide to unify `sync_committee_period_update` and `import_finalized_header` in SNO-446.